### PR TITLE
Address BinSkim complains by enabling Spectre mitigation

### PR DESCRIPTION
**Pull request by alexlamtest**
1. Updated VisualStudio2022.vsconfig to 
a) include components built with Spectre mitigation enabled, to address BinSkim's concern that our binaries are linking with vulnerable static libs;
b) drop components not required to build the 3 solutions (x64/release) on a local developer machine: WindowsAppRuntime.sln, MrtCore.sln, and WindowsAppRuntimeInstall.sln. The original components list with 87 items is now shortened by 18, while adding Spectre support. This saves resources on developer machines. 

2. Updated WindowsAppSDK.Build.Cpp.props to enable Spectre mitigation, to address BinSkim's concern.
3. Deleted VisualStudio2019.vsconfig because going forward we are dropping VS2019 support.
This PR is meant to address the "developer building WinAppSDK on local machine" scenario.
Addition changes may be required to silent BinSkim in the Azure pipeline built scenario.

How tested:

1. On a clean VM running a Win11 build, mimicking a developer's machine, deleted VisualStudio2019.vsconfig.
2. Grow VisualStudio2022.vsconfig from zero until the minimal set of VS2022 components is known in order to build the 3 solutions cited above. 
3. Reproduced the BinSkim complains about enabling Spectre mitigation by manually installing BinSkim and triggering a scan against the .dll and .exe (x64/release) binaries produced by the 3 solutions above. BinSkim spewed out dozens of complaints.
5. Applied the remaining changes in this PR.
6. Verified that a BinSkim re-scan no longer spewed out any complains about enabling Spectre mitigation.

### CODE REVIEW

**Review by kythant — APPROVED** [review]

File: docs/Coding-Guidelines/VisualStudio2022.vsconfig [hunk 1]
```diff
@@ -37,20 +35,16 @@
     "Component.Microsoft.VisualStudio.LiveShare.2022",
     "Microsoft.VisualStudio.Component.IntelliCode",
     "Microsoft.VisualStudio.Component.ClassDesigner",
-    "Microsoft.VisualStudio.Component.CodeClone",
     "Microsoft.VisualStudio.Component.GraphDocument",
     "Microsoft.VisualStudio.Component.CodeMap",
-    "Microsoft.VisualStudio.Component.DependencyValidation.Enterprise",
     "Microsoft.VisualStudio.Component.VC.CoreIde",
-    "Microsoft.VisualStudio.Component.Windows10SDK",
```
Comment (DrusTheAxe) [inline docs/Coding-Guidelines/VisualStudio2022.vsconfig:45]: I see Windows10SDK.19041 below. What's this, vLatest?

**Review by DrusTheAxe — APPROVED** [review]

**Review by codendone — APPROVED** [review]
:shipit:

File: docs/Coding-Guidelines/VisualStudio2022.vsconfig (hunk 1 above)
Comment (alexlamtest) [reply docs/Coding-Guidelines/VisualStudio2022.vsconfig:45]: That's my guess too, 19041 exists in both the original and new version, its seems this version is specifically called for.

**Review by bpulliam — APPROVED** [review]

**Pull request merged**